app/models/book.rb

```diff
@@ -1,8 +1,8 @@
 class Book < ApplicationRecord
   belongs_to :publisher, optional: true
-  has_and_belongs_to_many :authors
-  has_and_belongs_to_many :categories
-  
+  has_and_belongs_to_many :authors, -> { distinct }
+  has_and_belongs_to_many :categories, -> { distinct }
```

Comment:
what is the purpose of `distinct`?

Reply:
- Prevent duplicates if it doesn't exist book maybe have a list of duplicate authors

Reply:
I thought the in between table has both ids as primary key so it won't allow duplicates by default

Reply:
I tested it. I think `has_and_belongs_to_many` doesn't create an in-between table.